python/ray/autoscaler/command_runner.py

```diff
@@ -75,11 +75,14 @@ def remote_shell_command_str(self) -> str:
         """Return the command the user can use to open a shell."""
         raise NotImplementedError
 
-    def run_init(self, *, as_head: bool, file_mounts: Dict[str, str]) -> None:
+    def run_init(self, *, as_head: bool,
+                 file_mounts: Dict[str, str]) -> Optional[bool]:
         """Used to run extra initialization commands.
 
         Args:
             as_head (bool): Run as head image or worker.
             file_mounts (dict): Files to copy to the head and worker nodes.
+        Returns:
+            optional (bool): Whether initialization was run.
```

Comment:
@AmeerHajAli This is an early draft of a PR to allow changing file mounts with Docker & also to help fix stop (but not terminate) with Docker. Before I go any farther with this approach, I wanted to see if this change in CommandRunner is acceptable?

Reply:
Yes, bit you need to modify the function here too:
https://github.com/anyscale/product/blob/8462c65ff0e3a2f04e915e3d914e14634cdbae3c/backend/server/services/gateway_commandrunner_service.py#L86
1) edit the type annotation.
2) make it return `self.provider.get_response(request)`